src/lib/optimizer/abstract_syntax_tree/union_node.hpp

```diff
@@ -17,6 +19,20 @@ class UnionNode : public AbstractASTNode {
 
   std::string get_verbose_column_name(ColumnID column_id) const override;
 
+  const std::vector<std::string>& output_column_names() const override;
```

Comment:
Why does `UnionNode` have to override all of these? The left and right side are structurally identical (and this should be checked when the tree is changed, not when we call one of these methods).

Reply:
Needs to override because the default implementations in AbstractASTNode assume (and rightly so!) that the node only has a left input.

Reply:
Can't do the layout check in `_on_child_changed()` because when `_on_child_changed()` is called, a subtree might still be incomplete and calling `left_child()->output_column_names()` would crash.